fedimint-dbtool/src/main.rs

```diff
@@ -0,0 +1,73 @@
+use anyhow::Result;
+use bitcoin_hashes::hex::ToHex;
+use bytes::Bytes;
+use clap::{Parser, Subcommand};
+use fedimint_api::db::Database;
+use fedimint_api::module::registry::ModuleDecoderRegistry;
+
+#[derive(Debug, Clone, Parser)]
+struct Options {
+    database: String,
+    #[command(subcommand)]
+    command: DbCommand,
+}
+
+#[derive(Debug, Clone, Subcommand)]
+enum DbCommand {
+    List {
+        #[arg(value_parser = hex_parser)]
+        prefix: Bytes,
+    },
+    Write {
+        #[arg(value_parser = hex_parser)]
+        key: Bytes,
+        #[arg(value_parser = hex_parser)]
+        value: Bytes,
+    },
+    Delete {
+        #[arg(value_parser = hex_parser)]
+        prefix: Bytes,
+    },
+}
+
+fn hex_parser(hex: &str) -> Result<Bytes> {
+    let bytes: Vec<u8> = bitcoin_hashes::hex::FromHex::from_hex(hex)?;
+    Ok(bytes.into())
+}
+
+async fn open_db(path: &str) -> Result<Database> {
+    let rocksdb = fedimint_rocksdb::RocksDb::open(path)?;
+    Ok(Database::new(rocksdb, ModuleDecoderRegistry::default()))
+}
+
+fn print_kv(key: &[u8], value: &[u8]) {
+    println!("{} {}", key.to_hex(), value.to_hex());
+}
+
+#[tokio::main]
+async fn main() {
```

Comment:
Should we merge the functionality of this and dbdump so we only have 1 binary? I need to do some refactoring of dbdump anyway

Reply:
Eventually maybe when you have time. For now that works. I'd also like to discuss modularization of the DB tools. This one doesn't have to care yet fortunately xD